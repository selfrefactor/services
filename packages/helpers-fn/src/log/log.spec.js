const { log } = require('./log')

let logSpy

beforeEach(() => {
  logSpy = jest.spyOn(global.console, 'log')
})
afterEach(() => {
  logSpy.mockRestore()
})

const str = 'FOO bar'

test('separators', () => {
  log('separator')
  log('separatorx')
  expect(logSpy.mock.calls).toMatchSnapshot()
})

test('obj', () => {
  log({
    a : 1,
    b : [ 1, 2 ],
  },
  'obj')
  expect(logSpy.mock.calls[ 0 ]).toMatchSnapshot()
})

test('obj - when input is not an object', () => {
  log(str, 'obj')
  expect(logSpy.mock.calls[ 0 ]).toMatchSnapshot()
})

test('with wrong mode', () => {
  expect(() => log(str, 'not exist')).toThrow()
})

test('with too many inputs', () => {
  const expectedSpyCalls = [
    1,
    2,
    3,
    'helpers-fn.log doesn\'t support multiple inputs',
  ]
  log(
    1, 2, 3
  )
  expect(logSpy.mock.calls[ 0 ]).toEqual(expectedSpyCalls)
})
