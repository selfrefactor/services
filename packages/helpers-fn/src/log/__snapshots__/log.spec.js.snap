// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`obj - when input is not an object 1`] = `
[
  "FOO bar",
]
`;

exports[`obj 1`] = `
[
  "{
  a: [36m1[39m,
  b: [
    [36m1[39m,
    [36m2[39m
  ]
}",
]
`;

exports[`separators 1`] = `
[
  [
    "[38;2;71;59;123m_[39m[38;2;70;62;125m_[39m[38;2;70;64;126m_[39m[38;2;69;67;128m_[39m[38;2;68;70;130m_[39m[38;2;68;73;131m_[39m[38;2;67;75;133m_[39m[38;2;66;78;134m_[39m[38;2;66;81;136m_[39m[38;2;65;83;138m_[39m[38;2;64;86;139m_[39m[38;2;64;89;141m_[39m[38;2;63;91;143m_[39m[38;2;62;94;144m_[39m[38;2;62;97;146m_[39m[38;2;61;100;147m_[39m[38;2;60;102;149m_[39m[38;2;60;105;151m_[39m[38;2;59;108;152m_[39m[38;2;58;110;154m_[39m[38;2;58;113;156m_[39m[38;2;57;116;157m_[39m[38;2;56;118;159m_[39m[38;2;56;121;160m_[39m[38;2;55;124;162m_[39m[38;2;54;127;164m_[39m[38;2;54;129;165m_[39m[38;2;53;132;167m_[39m[38;2;53;135;168m_[39m[38;2;53;138;169m_[39m[38;2;52;141;170m_[39m[38;2;52;144;170m_[39m[38;2;52;146;171m_[39m[38;2;52;149;172m_[39m[38;2;52;152;173m_[39m[38;2;52;155;174m_[39m[38;2;51;158;175m_[39m[38;2;51;161;176m_[39m[38;2;51;164;176m_[39m[38;2;51;167;177m_[39m[38;2;51;170;178m_[39m[38;2;50;172;179m_[39m[38;2;50;175;180m_[39m[38;2;50;178;181m_[39m[38;2;50;181;181m_[39m[38;2;50;184;182m_[39m[38;2;49;187;183m_[39m[38;2;49;190;184m_[39m[38;2;49;193;185m_[39m[38;2;49;196;186m_[39m[38;2;49;198;187m_[39m[38;2;49;201;187m_[39m[38;2;48;204;188m_[39m[38;2;48;207;189m_[39m[38;2;48;210;190m_[39m",
  ],
  [
    "[38;2;71;59;123mðŸ€°[39m[38;2;71;60;124mðŸ€°[39m[38;2;70;62;125mðŸ€°[39m[38;2;70;63;125mðŸ€°[39m[38;2;70;64;126mðŸ€°[39m[38;2;69;66;127mðŸ€°[39m[38;2;69;67;128mðŸ€°[39m[38;2;69;68;129mðŸ€°[39m[38;2;68;70;130mðŸ€°[39m[38;2;68;71;130mðŸ€°[39m[38;2;68;73;131mðŸ€°[39m[38;2;67;74;132mðŸ€°[39m[38;2;67;75;133mðŸ€°[39m[38;2;67;77;134mðŸ€°[39m[38;2;66;78;134mðŸ€°[39m[38;2;66;79;135mðŸ€°[39m[38;2;66;81;136mðŸ€°[39m[38;2;65;82;137mðŸ€°[39m[38;2;65;83;138mðŸ€°[39m[38;2;65;85;138mðŸ€°[39m[38;2;64;86;139mðŸ€°[39m[38;2;64;87;140mðŸ€°[39m[38;2;64;89;141mðŸ€°[39m[38;2;63;90;142mðŸ€°[39m[38;2;63;91;143mðŸ€°[39m[38;2;63;93;143mðŸ€°[39m[38;2;62;94;144mðŸ€°[39m[38;2;62;96;145mðŸ€°[39m[38;2;62;97;146mðŸ€°[39m[38;2;61;98;147mðŸ€°[39m[38;2;61;100;147mðŸ€°[39m[38;2;61;101;148mðŸ€°[39m[38;2;60;102;149mðŸ€°[39m[38;2;60;104;150mðŸ€°[39m[38;2;60;105;151mðŸ€°[39m[38;2;59;106;152mðŸ€°[39m[38;2;59;108;152mðŸ€°[39m[38;2;59;109;153mðŸ€°[39m[38;2;58;110;154mðŸ€°[39m[38;2;58;112;155mðŸ€°[39m[38;2;58;113;156mðŸ€°[39m[38;2;57;114;156mðŸ€°[39m[38;2;57;116;157mðŸ€°[39m[38;2;57;117;158mðŸ€°[39m[38;2;56;118;159mðŸ€°[39m[38;2;56;120;160mðŸ€°[39m[38;2;56;121;160mðŸ€°[39m[38;2;55;123;161mðŸ€°[39m[38;2;55;124;162mðŸ€°[39m[38;2;55;125;163mðŸ€°[39m[38;2;54;127;164mðŸ€°[39m[38;2;54;128;165mðŸ€°[39m[38;2;54;129;165mðŸ€°[39m[38;2;53;131;166mðŸ€°[39m[38;2;53;132;167mðŸ€°[39m",
  ],
]
`;
